### ISSUE TITLE
Export to GNPS of MS1_MS2 data

### ISSUE THREAD

**Issue by russof85** [mzmine/mzmine2#926]
### Basic information

* My operating system: CentOS Linux 7 (Core)
* My MZmine version: MZmine-2.37.1.corr17.7

### What happened
Hi! I'm analyzing a large amount of data with MZmine (more than 2000 samples; metabolomics data) in batch mode and all the steps worked very well (after some fights with memory issues). The last two steps consist of the 'Export to GNPS' module. One step to export only the MS2 data and another step to export all the data. For the MS2 data worked well and it took just few minutes, but for all the data is taking a long time. It's running since 1 day and 13 hours without reporting any error. I show the screenshots of the log and the xml batch files anyway.
<img width="886" alt="MZmine_export_to_GNPS_MS1_MS2_log" src="https://user-images.githubusercontent.com/73115906/127318331-bfb7d286-2a34-42d2-a625-bc28932a489d.PNG">
<img width="634" alt="MZmine_export_to_GNPS_MS1_MS2_xml_batch_file" src="https://user-images.githubusercontent.com/73115906/127320027-6b63360a-4076-42ba-bfba-87a0d538a2a5.PNG">

Everything worked fine during the optimization of the parameters with a small number of samples. I can see that some of the steps in MZmine require a large amount of memory (e.g. metaCorrelate) especially with this huge amount of data, but the server I'm working on right now was able to handle it by using 128GB of RAM (Java was crashing with less than that). 
Do you think  the 'Export to GNPS' of MS1_MS2 data is requiring a lot of RAM and reaching the Java heap_size? (currently set to 128GB).

I know that it's not easy to know what's going on but any comment/suggestion is very much appreciated! 

Thanks a lot!

Francesco

**Comment by robinschmid** [comment]
Hey Francesco,

interesting question. 
About the memory - we are trying to improve that in MZmine 3.
You can also try to check "remove original feature list" in every step to free memory from unused feature lists (e.g., the chromatograms after feature deconvolution)

So the export to FBMN on GNPS worked - but the export of ion identity molecular networking did not?
We had a similar problem once where the number of concurrent threads was not high enough and this led to the tasks blocking the last one from finishing. But that's only reasonable if the threads are <=4. (see mzmine preferences)

The export should not use too much memory or anything so hart to tell what happens here. 

Maybe try setting the filter to "MS2 or annotation" to only include features with MS2 or ion annotation. All the other features are by default disregarded in GNPS

**Comment by russof85** [comment]
Hi Robin! 

Thanks for your reply. Yes, it would be amazing to improve the memory issues but I see the challenges! I have removed the original feature list in every step indeed to avoid any memory issue.
I have selected 'MS2 or annotation' to export the MS2 data for GNPS as you said and it works well, but it doesn't export anything when 'ALL' is selected in the filters (I need both MS1 and MS2 data for my downstream analyses and I just wanted to use the 'Export to GNPS' module as well). It worked well with the small sample size during parameter optimization though.  Maybe I can experiment with other export functions for exporting the full data using for instance 'Export to csv file' or similar. Let's see if it works :) 

Francesco
